### PR TITLE
Issue #38: Implement endpoints related to Processes

### PR DESCRIPTION
**Pull request by nadre**
This pull request contains the implementation of the new endpoints as described in Issue #38

### CODE REVIEW

**Comment by Philly-B** [comment]
Test

**Comment by reckart** [comment]
Please run "make black" for auto-formatting.

**Comment by reckart** [comment]
I have changed the target branch from main to dev - PRs should go to dev. Main should always be the latest release version... I also make the mistake sometimes.... we need to pay attention.

Can you please merge the latest change from dev into your branch, resolve conflicts, `make black` and then re-request the review. Thanks!

**Review by reckart — CHANGES_REQUESTED** [review]
Many method do not have pydoc.

File: averbis/core/_rest_client.py [hunk 1]
```diff
@@ -705,16 +741,43 @@ def install_pear(self, file_or_path: Union[IO, Path, str]) -> Pear:
...
+    def get_process(self, process_name: str, document_source_name: str) -> Process:
+        """
+        Get a process
+        :return: The process
+        """
+        # noinspection PyProtectedMember
+        return self.client._get_process(self, process_name, document_source_name)
+
+    @experimental_api
+    def list_processes(self) -> List[Tuple[str, str]]:
```
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py:794]: We thought about this one too. A List[Process] totally makes sense, it just requires another backend call for each process. But that should not be a problem.

File: averbis/core/_rest_client.py [hunk 2]
```diff
@@ -510,21 +512,51 @@ def __init__(self, project: "Project", identifier: str):
...
         )
 
 
+class Process:
+    def __init__(self,
+                 project: "Project",
+                 name: str,
+                 document_source_name: str,
+                 pipeline_name: str,
+                 state: str,
```
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py]: I totally see the point but I'm not sure if it is worth opening the gap between client and product api.

File: averbis/core/_rest_client.py [hunk 3]
```diff
@@ -510,21 +512,51 @@ def __init__(self, project: "Project", identifier: str):
...
+        self.processed_documents = processed_documents
+
+    def delete(self):
+        """
+        Deletes the Process.
+        """
+        # noinspection PyProtectedMember
+        self.project.client._delete_process(self.project.name, self.name, self.document_source_name)
+
+    def reprocess(self):
```
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py:569]: The process starts as soon as it is created. stop() is not implemented yet.

File: averbis/core/_rest_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py:794]: Why does it require another backend call for each process? I'd expect that a `list_processes` REST API endpoint provides sufficient information to instantiate a `Process` python object.

File: averbis/core/_rest_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py]: Yes, totally :) You can also change the REST API if you like :P

File: averbis/core/_rest_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py:569]: Can we have a `stop()`? 🥺

[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py:569]: Suppose I'd like to hack myself a `stop()` - what happens if I do this?

```
process = project.createProcess("my-process")
process.start()
process.delete()
process = project.createProcess("my-process")
```

File: averbis/core/_rest_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py]: Actually, we already have a two-layer architecture in the Python API.

At the low level is the `Client` which more-or-less resembles exactly the endpoints as provided by the product. 
At the high level, we have the convenience objects like `Project` or `Process` which aim to make the whole thing more user-friendly for the Python programmer. So we do have a place for that gap you mention already and use it - intentionally.

File: averbis/core/_rest_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py:569]: I find it very strange/inconvenient that a process is started immediately upon creation. I'd expect that the process is created and then can be started separately.

[the remark this answers is not in the document]
Comment (reckart) [reply averbis/core/_rest_client.py:569]: At the very least, I would then expect that `project.create_process()` rather be called`project.start_new_process()`.

File: averbis/core/_rest_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py]: done.

File: averbis/core/_rest_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py]: done.

File: averbis/core/_rest_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nadre) [reply averbis/core/_rest_client.py:571]: out of scope.

**Comment by nadre** [comment]
@reckart & @Philly-B I think I'm done here. Please check and merge if everything's fine.

**Review by reckart — APPROVED** [review]

**Comment by reckart** [comment]
@Philly-B fine by me - would you like to give a thumbs up as well and merge?

**Pull request merged**